core/src/main/java/org/testcontainers/utility/DockerImageName.java

```diff
@@ -124,8 +144,13 @@ public String getRegistry() {
         return registry;
     }
 
+    public DockerImageName withTag(final String newTag) {
```

Comment:
Love it!

Made me thinking whether we should even have a fully featured builder as well:
```java
DockerImageName.builder()
    .registry("repo.example.com")
    .image("services/foo")
    .tag("123")
    .build();
```

Reply:
I feel like the halfway house of `Dockerfilename.of("repo.example.com/services/foo").withTag("123")` is probably enough - I'd like to try that first.

Reply:
FTR the builder idea is just an idea, don't need to be in this PR and can easily be a follow up :)